Minio/DataModel/AccessCredentials.cs

```diff
@@ -26,9 +26,14 @@ public class AccessCredentials
     public AccessCredentials(string accessKey, string secretKey,
         string sessionToken, DateTime expiration)
     {
-        if (string.IsNullOrWhiteSpace(accessKey) || string.IsNullOrWhiteSpace(secretKey))
-            throw new ArgumentNullException(nameof(accessKey) + " and " + nameof(secretKey) +
-                                            " cannot be null or empty.");
+        if (string.IsNullOrEmpty(accessKey))
+            throw new ArgumentException($"'{nameof(accessKey)}' cannot be null or empty.", nameof(accessKey));
+
+        if (string.IsNullOrEmpty(secretKey))
+            throw new ArgumentException($"'{nameof(secretKey)}' cannot be null or empty.", nameof(secretKey));
+
+        if (string.IsNullOrEmpty(sessionToken))
+            throw new ArgumentException($"'{nameof(sessionToken)}' cannot be null or empty.", nameof(sessionToken));
```

Comment:
I think there is a logic mistake here.
We don't need all 3 parameters to be provided all together, i.e. either `accessKey` and `secretKey` or `sessionToken` and `Expiration` need to be provided. 

So, any one of the pair is good for access to be granted. Of course at least one set should to be required.
I'll also double check Go SDK to make sure it is implemented this way.